mlx/ops.h

```diff
@@ -1105,4 +1105,26 @@ array addmm(
     const float& beta = 1.f,
     StreamOrDevice s = {});
 
+/** Implements the identity function but allows injecting dependencies to other
+ * arrays to ensure that they will be computed before anything that depends on
+ * the inputs. */
+std::vector<array> depends(
+    const std::vector<array>& inputs,
+    const std::vector<array>& dependencies);
+
+/** Allows to have call a function and define a custom VJP for it. */
+std::vector<array> custom_vjp(
+    std::function<std::vector<array>(const std::vector<array>&)> fun,
+    std::function<std::vector<array>(
+        const std::vector<array>&,
+        const std::vector<array>&,
+        const std::vector<array>&)> fun_vjp,
+    const std::vector<array>& args);
```

Comment:
this is super cool! custom vjps ✅

Reply:
I would probably say this also is more like a transform than an op though. We may also just want a new file..

It's composable right? Like I could wrap a custom vjp of a function in another function and make a custom vjp (or transform it)?

Reply:
Transformations like getting the vjp of course works. Vmap should be easy but I didn't implement it. We just need to vmap the passed functions.

Yeah I think both this and checkpoint should go in transforms.